c_glib/arrow-glib/array-builder.cpp

```diff
@@ -4486,6 +4489,262 @@ garrow_struct_array_builder_get_field_builders(GArrowStructArrayBuilder *builder
+ *
+ * Returns: %TRUE on success, %FALSE if there was an error.
+ *
+ * Since: 1.0.0
+ */
+gboolean
+garrow_map_array_builder_append_null(GArrowMapArrayBuilder *builder,
+                                     GError **error)
+{
+  return garrow_array_builder_append_null<arrow::MapBuilder *>
+    (GARROW_ARRAY_BUILDER(builder),
+     error,
+     "[map-array-builder][append-null]");
+}
+
+/**
+ * garrow_map_array_builder_append_nulls:
+ * @builder: A #GArrowMapArrayBuilder.
+ * @n: The number of null values to be appended.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Append multiple nulls at once. It's more efficient than multiple
+ * `append_null()` calls.
+ *
+ * Returns: %TRUE on success, %FALSE if there was an error.
+ *
+ * Since: 1.0.0
+ */
+gboolean
+garrow_map_array_builder_append_nulls(GArrowMapArrayBuilder *builder,
+                                      gint64 n,
+                                      GError **error)
+{
+  return garrow_array_builder_append_nulls<arrow::MapBuilder *>
+    (GARROW_ARRAY_BUILDER(builder),
+     n,
+     error,
+     "[map-array-builder][append-nulls]");
+}
+
+/**
+ * garrow_map_array_builder_get_key_builder:
+ * @builder: A #GArrowMapArrayBuilder.
+ *
+ * Returns: (transfer none): The #GArrowArrayBuilder for key values.
+ *
+ * Since: 1.0.0
+ */
+GArrowArrayBuilder *
+garrow_map_array_builder_get_key_builder(GArrowMapArrayBuilder *builder)
+{
+  auto priv = GARROW_MAP_ARRAY_BUILDER_GET_PRIVATE(builder);
+  if (!priv->key_builder) {
+    auto arrow_builder =
+      static_cast<arrow::MapBuilder *>(
+        garrow_array_builder_get_raw(GARROW_ARRAY_BUILDER(builder)));
+    auto arrow_key_builder = arrow_builder->key_builder();
+    priv->key_builder = garrow_array_builder_new_raw(arrow_key_builder);
+    garrow_array_builder_release_ownership(priv->key_builder);
+  }
+  return priv->key_builder;
+}
+
+/**
+ * garrow_map_array_builder_get_item_builder:
+ * @builder: A #GArrowMapArrayBuilder.
+ *
+ * Returns: (transfer none): The #GArrowArrayBuilder for item values.
+ *
+ * Since: 1.0.0
+ */
+GArrowArrayBuilder *
+garrow_map_array_builder_get_item_builder(GArrowMapArrayBuilder *builder)
+{
+  auto priv = GARROW_MAP_ARRAY_BUILDER_GET_PRIVATE(builder);
+  if (!priv->item_builder) {
+    auto arrow_builder =
+      static_cast<arrow::MapBuilder *>(
+        garrow_array_builder_get_raw(GARROW_ARRAY_BUILDER(builder)));
+    auto arrow_item_builder = arrow_builder->item_builder();
+    priv->item_builder = garrow_array_builder_new_raw(arrow_item_builder);
+    garrow_array_builder_release_ownership(priv->item_builder);
+  }
+  return priv->item_builder;
+}
+
+/**
+ * garrow_map_array_builder_get_value_builder:
+ * @builder: A #GArrowMapArrayBuilder.
+ *
+ * Returns: (transfer none): The #GArrowArrayBuilder for struct values.
+ *   This is used instead of garrow_map_array_builder_get_key_builder() and garrow_map_array_builder_get_item_builder().
```

Comment:
How about this?

```cpp
  * Returns: (transfer none): The #GArrowArrayBuilder to add map entries as struct values.
  *   This can be used instead of garrow_map_array_builder_get_key_builder() and
  *   garrow_map_array_builder_get_item_builder(). You can build map entries as a list of
  *   struct values with this builder.
```

Reply:
Thank you for improving document. I've applied this.